src/ui/browserAction/popupPage.js

```diff
@@ -0,0 +1,294 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import {
+  html,
+  css,
+  LitElement,
+  render,
+  createRef,
+  ref,
+} from "../../vendor/lit-all.min.js";
+
+import { fontSizing, resetSizing } from "../../components/styles.js";
+
+// Other components used
+import "./../../components/stackview.js";
+import "./../../components/serverlist.js";
+import "./../../components/vpncard.js";
+import "./../../components/titlebar.js";
+
+/**
+ * @typedef {import("../../background/vpncontroller/states.js").VPNState} VPNState
+ */
+
+/**
+ * This is the Component for the Browser Action Popup. #
+ *
+ * It will establish a connection to the backend script
+ * feed data downwards to the components and send component events upwards
+ * to the backend script.
+ */
+export class BrowserActionPopup extends LitElement {
+  static properties = {
+    vpnState: { type: Object },
+    pageURL: { type: String },
+  };
+
+  constructor() {
+    super();
+    this.pageURL = "about:blank";
+
+    // Hook up the Controller Port to check the VPN Status
+    // and get the ServerList
+    /** @type {VPNState} */
+    this.vpnState = null;
+    this.controllerPort = globalThis.chrome.runtime.connect({
+      name: "vpncontroller",
+    });
+    this.controllerPort.onMessage.addListener((state) => {
+      console.log(state);
+      this.vpnState = state;
+    });
+
+    // Get the Page URL that is active at the time of opening the popup
+    chrome.tabs.query({ active: true, currentWindow: true }, (tabs) => {
```

Comment:
For the pageAction icon and menu we'll have to 1.) get the hostname of the active tab on every active tab change and then 2.) check if a siteContext exists for that hostname. It'd be nice if we could just cache and return that info via a port (or whatever) so that the popup does not have to do the work again on open. Another maybe nice thing about getting this info from the background is we can use the same url formatting helper in case we need to strip out `about:reader` prefixes.

Reply:
I think that is fair. however this component still is "dumb" and that is no problem for me. i mostly wanted to demo that it can adjust size depending on what we render and we have some switch :)